web/server/api/endpoints/files.py

```diff
@@ -46,21 +46,22 @@ def get_file(
     """Get a file, including its contents."""
     try:
         file_path = Path(path)
-        file = _get_file_with_content(file_path, settings)
+        file = _get_file_with_content(settings.project_path / file_path, str(file_path))
     except FileNotFoundError:
         raise HTTPException(status_code=HTTP_404_NOT_FOUND)
 
     return file
 
 
-@router.post("/{path:path}", response_model=models.File)
+@router.post("/{path:path}", response_model=t.Optional[models.File])
 async def write_file(
+    response: Response,
     content: str = Body("", embed=True),
     new_path: t.Optional[str] = Body(None, embed=True),
     path: str = Depends(validate_path),
     settings: Settings = Depends(get_settings),
     context: Context = Depends(get_context),
-) -> models.File:
+) -> t.Optional[models.File]:
```

Comment:
This can no longer return a File object. Can the return type be just None?

Reply:
`File` may not be picked up by openapi to convert to typescript types if it is not exported